lib/chai/core/assertions.js

```diff
@@ -988,6 +988,13 @@ module.exports = function (chai, _) {
 
   function assertInstanceOf (constructor, msg) {
     if (msg) flag(this, 'message', msg);
+    var isFunction = constructor instanceof Function;
+
+    if (!isFunction) {
```

Comment:
Oops, speaking of edge cases, `typeof` is not sufficient here.

`instanceof` works not only with functions, but with any objects that have `@@hasInstance`.